arelle/plugin/validate/DBA/rules/fr.py

```diff
@@ -456,6 +458,58 @@ def rule_fr56(
             )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+)
+def rule_fr74a(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    DBA.FR74a: Provisions (fsa:Provisions) must be less than or equal to the balance sheet total (fsa:LiabilitiesAndEquity)
+    minus equity (fsa:Equity).
+    """
+    groupedFacts = getFactsGroupedByContextId(val.modelXbrl, pluginData.equityQn, pluginData.liabilitiesQn, pluginData.provisionsQn, pluginData.liabilitiesOtherThanProvisionsQn)
+    for contextID, facts in groupedFacts.items():
+        equityFact = None
+        liabilityFact = None
+        liabilityOtherFact = None
+        provisionFact = None
+        for fact in facts:
+            if fact.qname == pluginData.equityQn and fact.unit.value == DANISH_CURRENCY:
+                equityFact = fact
+            elif fact.qname == pluginData.liabilitiesQn and fact.unit.value == DANISH_CURRENCY:
+                liabilityFact = fact
+            elif fact.qname == pluginData.provisionsQn and fact.unit.value == DANISH_CURRENCY:
+                provisionFact = fact
+            elif fact.qname == pluginData.liabilitiesOtherThanProvisionsQn and fact.unit.value == DANISH_CURRENCY:
+                liabilityOtherFact = fact
+        if equityFact is not None and liabilityFact is not None and provisionFact is not None:
+            if cast(float, liabilityFact.xValue) - cast(float, equityFact.xValue) < cast(float, provisionFact.xValue) - ROUNDING_MARGIN:
```

Comment:
Are these actually `xs:float`, or would `decimal.Decimal` be better?

Reply:
These xValues are actually floats.

Reply:
The `LiabilitiesAndEquity` concept is `xbrli:monetaryItemType`, which extends `xs:decimal`.  I believe [this logic](https://github.com/Arelle/Arelle/blob/9bde72ef61ff834d562468070580af79769d1468/arelle/XmlValidate.py#L440-L443) should be producing `decimal.Decimal`.  When I run `fr74a-invalid.xbrl`, I see `<class 'decimal.Decimal'>` for `type(liabilityFact.xValue)`, not `<class 'float'>`, so `cast(float)` seems wrong.